### ISSUE TITLE
[Bug]: TSE Sender Mismatch false positive?

### ISSUE THREAD

**Issue by ghost** [YimMenu/YimMenu#3195]
### Describe the bug

I wanted to ask, have you ever experienced this: when you play for several hours continuously, sometimes switching sessions a couple of times, you start receiving TSE Sender Mismatch from random players, and after a few seconds, you start getting player removal crash from everyone, and the game crashes?

### Steps To Reproduce

1. Join the session for a few hours
2. You can sometimes change the session
3. Wait until you start receiving notifications about "Received TSE Sender Mismatch" from random players
4. Wait for approximately 10-20 seconds and get a crash(Player removal crash) from all players or the nearest ones

### Expected Behavior

Everything should be fine

### Logs

Sorry for the dirty build, I was just testing other pull requests, but the same thing happened on the main branch too. If it sounds unconvincing, I can later do the same with the official build
[cout.log](https://github.com/YimMenu/YimMenu/files/15469453/cout.log)

### Store

Steam

### Language

Russian

### YimMenu Language

Русский (RU)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by Rxann** [comment]
I am guessing it's not a false positive because well it is probably being sent as part of the crash.

**Comment by Mr-X-GTA** [comment]
It looks like the game crashed because all player peds (including the local one) were deleted

**Comment by Rxann** [comment]
> It looks like the game crashed because all player peds (including the local one) were deleted

then that means we aren't blocking the player ped removal crash correctly(I think).

**Comment by maybegreat48** [comment]
I have a few ideas on what this is, but I need some logs to confirm my suspicions. Can you enable "Settings > Reactions > Notifications > Player Join > Log" and "Player Leave > Log"? Can you also enable "Settings > Debug > Logs > Log Packets > Filters"? The next time this happens to you, send your cout.log and packets.log

Did you log packets at the time of this attack? Do you have more info (like did players leave shortly before the crash)? Do you also have the list of players that were in this session at that time?

**Comment by maybegreat48** [comment]
Also, was the host a modder? Do you have them in your database?

> It looks like the game crashed because all player peds (including the local one) were deleted

The delete requests are likely just a side-effect of the method used to induce this crash (which I choose not to reveal in public)

**Comment by ghost** [comment]
> Also, was the host a modder? Do you have them in your database?
> 
> > It looks like the game crashed because all player peds (including the local one) were deleted
> 
> The delete requests are likely just a side-effect of the method used to induce this crash (which I choose not to reveal in public)

No, I was the session host.  if you need rid of those people who sent me Tse Mismatch then they are in the log file

**Comment by maybegreat48** [comment]
Does this happen often, or is this a one time thing? Also, *when* did this happen? Yesterday?

**Comment by ghost** [comment]
> Does this happen often, or is this a one time thing? Also, _when_ did this happen? Yesterday?

this happened 4-5 times.  One day I decided to change the session after I started getting Tse Mismatch but after I joined a new session, after a minute it started again and I ended up crashing again.  Perhaps this is caused by one of these?  I always have a script running that assigns a bounty to everyone in the session, sometimes I spawn xml vehicles.  about when this happened, the last time it happened was today.  but this happened last week too, well, at least this didn’t happen before. Perhaps the crash occurs due to the fact that when we start receiving ped removal crash from everyone in the session at the same time, then the yimmenu cannot cope with this and when I see a notification about crashes from everyone at the same time but it is not displayed in cout

**Comment by maybegreat48** [comment]
What I think is that this isn't random and is actually a targeted attack. I managed to reproduce the first part of this crash and got everyone's TSEs to mismatch for players in a session. Do you get the player remove sync every time this happens to you? Can you send me the logs (found in the backups)?

**Comment by ghost** [comment]
> What I think is that this isn't random and is actually a targeted attack. I managed to reproduce the first part of this crash and got everyone's TSEs to mismatch for players in a session. Do you get the player remove sync every time this happens to you? Can you send me the logs (found in the backups)?

do you mean do I disable player sync?  do you mean timeout?  The first time this happened for me, I had an autokick and a timeout, and I kicked out everyone from whom I received Tse Mismatch.  but the crash still happened after that, I half thought that it was just someone blaming the players for this and I didn’t set a timeout or kick, but just waited and still, after endless spam of these notifications, after 30-40 seconds the crash still happened. Regarding backups, before that I played a little with the code and got crashes of hundreds of megabytes, and periodically I cleared the backups.  but now when I will play again I will include everything that u need (log packets, join player, log etc.) and I will send cout file and packet log file to you as soon as this happens again

**Comment by maybegreat48** [comment]
> I half thought that it was just someone blaming the players for this

That's not normal

> I kicked out everyone from whom I received Tse Mismatch. but the crash still happened after that

They are legit players. Kicking them won't make the crash go away

Were you always the host when this happened to you?
Also did your game do weird things before the crash? Like this?

![crash report](https://github.com/YimMenu/YimMenu/assets/96936658/2ee19649-c6e4-4817-93ee-c0f99749a1d3)

**Comment by lonelybud** [comment]
i have a log from my own fork recently. I didn't got crashed or anything but i received `TSE Sender Mismatch` - 
[05-26-2024-17-05-38_cout.log](https://github.com/YimMenu/YimMenu/files/15478788/05-26-2024-17-05-38_cout.log)

**Comment by ghost** [comment]
> > I half thought that it was just someone blaming the players for this
> 
> That's not normal
> 
> > I kicked out everyone from whom I received Tse Mismatch. but the crash still happened after that
> 
> They are legit players. Kicking them won't make the crash go away
> 
> Were you always the host when this happened to you? Also did your game do weird things before the crash? Like this?
> 
> ![crash report](https://private-user-images.githubusercontent.com/96936658/334582062-2ee19649-c6e4-4817-93ee-c0f99749a1d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY5NjExNzIsIm5iZiI6MTcxNjk2MDg3MiwicGF0aCI6Ii85NjkzNjY1OC8zMzQ1ODIwNjItMmVlMTk2NDktYzZlNC00ODE3LTkzZWUtYzBmOTk3NDlhMWQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI5VDA1MzQzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2NDAwNTU0YWFkNmJiYTQ3MTQyOWY0ZGE0MDhiYzdlODY1NWNiMTgzYzkyN2ZkNjYzMWU4OGNlYjQ2NDYyNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wwEHkNJVGQRlzJyOuR1LtNh0h__ktX6ZPpyn9pvYYy8)

Yes, I always stay only in those sessions where I was able to become the host.  no, my character didn’t disappear

**Comment by maybegreat48** [comment]
> i have a log from my own fork recently. I didn't got crashed or anything but i received `TSE Sender Mismatch` - [05-26-2024-17-05-38_cout.log](https://github.com/YimMenu/YimMenu/files/15478788/05-26-2024-17-05-38_cout.log)

Do you have packet logging when this happened?

**Comment by lonelybud** [comment]
> Do you have packet logging when this happened?

no

**Comment by maybegreat48** [comment]
@CringeArab when these crashes happened, were you playing with matchmaking parameters or otherwise set your region to China in any way? Some spammers seem to break sessions when they enter

@lonelybud how often does this happen to you?

**Comment by ghost** [comment]
> @CringeArab when these crashes happened, were you playing with matchmaking parameters or otherwise set your region to China in any way? Some spammers seem to break sessions when they enter
> 
> @lonelybud how often does this happen to you?

yes, I use the filter for selecting players by number and sometimes I connect to the Chinese this way
![pic](https://github.com/YimMenu/YimMenu/assets/168850586/8c743eea-8485-4760-a6d0-d4eed0a87afa)
If you have any more questions about the settings, here are
[mine](https://github.com/YimMenu/YimMenu/files/15495017/settings.json.txt)

**Comment by ghost** [comment]
so it happened again, I didn't do anything but just stood there afk. It started with the same thing that I started getting tse mismatch but in the end it didn’t crash me but I now noticed that I’m not showing up in the menu table. in cout I noticed that I was “Disconnected” although there are players in the menu table but they are nowhere on the map. Oh, and the map became like in single player. the session is empty from my point of view, although the number of players in the upper left corner of the menu changes
[cout.log](https://github.com/YimMenu/YimMenu/files/15495517/cout.log)
[packets.log](https://github.com/YimMenu/YimMenu/files/15495524/packets.log)
![pic1](https://github.com/YimMenu/YimMenu/assets/168850586/3c9d7e24-4ab2-43bd-830f-da880a2771f9)
![pic2](https://github.com/YimMenu/YimMenu/assets/168850586/fef65035-6bdb-48dd-a5e4-de86adcb8ff6)

**Comment by ghost** [comment]
Now I changed the session and after a couple of minutes I also received a mismatch and noticed that in the menu table there was a player named xX_THE_SWAT_Xx who was displayed in the table 2 times. but when I clicked on the first one, the player window did not open. Well, being the host, I kicked him through breakup and after that I accidentally died and endless loading began and again I am not in the players table in the menu. The menu again thinks that I'm disconnected `Player left '--muhammad--' freeing slot #1 with Rockstar ID:****` continuation of logs: [cout.log](https://github.com/YimMenu/YimMenu/files/15495763/cout.log) 
And packet log
[packets.zip](https://github.com/YimMenu/YimMenu/files/15495972/packets.zip)

**Comment by lonelybud** [comment]
This is a good investigation.

**Comment by maybegreat48** [comment]
Yeah, this is public now I guess. I'll add this "feature" and protections for it in my next PR, but note that certain variations of this would break sessions anyway and cannot be blocked (I'm not adding those)

**Comment by ghost** [comment]
Perhaps we should close this?

**Comment by maybegreat48** [comment]
We can keep this open for a while; the more data we get about this, the better. I want to see why the spammers are using this exploit

And there's more to patch
